nomad/structs/volumes.go

```diff
@@ -279,6 +281,43 @@ func (v *VolumeMount) Copy() *VolumeMount {
 	return nv
 }
 
+func (v *VolumeMount) Validate() error {
+	var mErr multierror.Error
+
+	// Validate the task does not reference undefined volume mounts
+	if v.Volume == "" {
+		mErr.Errors = append(mErr.Errors, errVolMountEmptyVol)
+	}
+
+	if !v.MountPropagationModeIsValid() {
+		mErr.Errors = append(mErr.Errors, fmt.Errorf("%w: \"%s\"", errVolMountInvalidPropagationMode, v.PropagationMode))
```

Comment:
The `%q` formatter is for just this case:

```suggestion
		mErr.Errors = append(mErr.Errors, fmt.Errorf("%w: %q", errVolMountInvalidPropagationMode, v.PropagationMode))
```